holoviews/core/data/image.py

```diff
@@ -0,0 +1,250 @@
+            data = np.flipud(data)
+            return data.T.flatten() if flat else data
+        else:
+            return None
+
+
+    @classmethod
+    def select(cls, dataset, selection_mask=None, **selection):
+        """
+        Slice the underlying numpy array in sheet coordinates.
+        """
+        selection = {k: slice(*sel) if isinstance(sel, tuple) else sel
+                     for k, sel in selection.items()}
+        coords = tuple(selection[kd.name] if kd.name in selection else slice(None)
+                       for kd in dataset.kdims)
+        if not any([isinstance(el, slice) for el in coords]):
+            return dataset.data[dataset.sheet2matrixidx(*coords)]
+
+        # Apply slices
+        xidx, yidx = coords
+        l, b, r, t = dataset.bounds.lbrt()
+        if isinstance(xidx, slice):
+            l = l if xidx.start is None else max(l, xidx.start)
+            r = r if xidx.stop is None else min(r, xidx.stop)
+        if isinstance(yidx, slice):
+            b = b if yidx.start is None else max(b, yidx.start)
+            t = t if yidx.stop is None else min(t, yidx.stop)
+        bounds = BoundingBox(points=((l, b), (r, t)))
+        slc = Slice(bounds, dataset)
+        data = slc.submatrix(dataset.data)
+        return data
+
+
+    @classmethod
+    def sample(cls, dataset, samples=[]):
+        """
+        Sample the Raster along one or both of its dimensions,
+        returning a reduced dimensionality type, which is either
+        a ItemTable, Curve or Scatter. If two dimension samples
+        and a new_xaxis is provided the sample will be the value
+        of the sampled unit indexed by the value in the new_xaxis
+        tuple.
+        """
+        if len(samples[0]) == 1:
+            return dataset.select(**{dataset.kdims[0].name:
+                                     [s[0] for s in samples]}).columns()
+        else:
+            return [c+(dataset.data[dataset._coord2matrix(c)],)
+                    for c in samples]
+
+    @classmethod
+    def length(cls, dataset):
+        return np.product(dataset.data.shape)
+
+
+    @classmethod
+    def groupby(cls, dataset, dim_names, container_type, group_type, **kwargs):
+        # Get dimensions information
+        dimensions = [dataset.get_dimension(d) for d in dim_names]
+        kdims = [kdim for kdim in dataset.kdims if kdim not in dimensions]
+
+        # Update the kwargs appropriately for Element group types
+        group_kwargs = {}
+        group_type = dict if group_type == 'raw' else group_type
+        if issubclass(group_type, Element):
+            group_kwargs.update(util.get_param_values(dataset))
+            group_kwargs['kdims'] = kdims
+        group_kwargs.update(kwargs)
+
+        if len(dimensions) == 1:
+            didx = dataset.get_dimension_index(dimensions[0])
+            coords = dataset.dimension_values(dimensions[0], False)
+            xvals = dataset.dimension_values(abs(didx-1), False)
```

Comment:
I forget what the ``False`` is about - showing the keyword might help me remember!